### PR TITLE
feat: add support for nix shell

### PR DESCRIPTION
**Pull request by Math-42**
Add support for the ephemeral nix shell.

How to test?

install [nix](https://nixos.org/download.html#nix-install-linux), enter the project directory and run:

nix-shell

then proceed to the bootstrap and build.

closes https://github.com/courselab/snaskii22/issues/5

### CODE REVIEW

File: shell.nix [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{ pkgs ? import <nixpkgs> {} }:
+
+with pkgs;
+
+mkShell {
+  nativeBuildInputs = [
+    ncurses5
```
Comment (gp2112) [inline shell.nix:7]: won't you need `ncurses5` to run the program? 

`buildInputs` are all packages you need to run the program and `nativeBuildInputs` are all packages you need to build it

**Comment by gp2112** [comment]
@Math-42 I've added nix with flakes and nix packaging :)

Now it's very easy to build it for anyone who uses nix :)

just: `nix run github:courselab/snaskii22/<branch_name>`!

[asciicast](https://asciinema.org/a/Sni5g2HRC8kNXCsIRs2J1q0hC)

You just substitute the end of URL with the repository's branch. 
If merged to develop, it would be `nix run github:courselab/snaskii22/nix`

**Review by CerqueiraMatheus — APPROVED** [review]
It looks great to me!

**Comment by CerqueiraMatheus** [comment]
@gp2112 please, also add the new nix build instructions to the README

File: README.md [hunk 2]
```diff
@@ -5,6 +5,11 @@ arcades of the 80s. It was meant to be developed as a collaborative
 programming exercise in a course on Open Source Systems taught to
 undergraduate CS students.
 
+
+## Building/Running with Nix
```
Comment (CerqueiraMatheus) [inline README.md:9]: It could be inside the Quick start section, following the existing pattern, e.g. "If you are using nix...".

File: README.md [hunk 3]
```diff
@@ -5,6 +5,11 @@ arcades of the 80s. It was meant to be developed as a collaborative
...
+
+`nix run github:courselab/snaskii22`
```
Comment (CerqueiraMatheus) [inline README.md:11]: You should put the code inside  ``` to match the existing code snippets.

**Review by CerqueiraMatheus — CHANGES_REQUESTED** [review]

**Review by CerqueiraMatheus — APPROVED** [review]
I fixed a small typo. Great work!

**Review by gp2112 — APPROVED** [review]

**Comment by gp2112** [comment]
@CerqueiraMatheus I'll just squash the last commit, than I'll merge

**Review by CerqueiraMatheus — APPROVED** [review]

**Comment by CerqueiraMatheus** [comment]
Go ahead!

**Pull request merged**